### ISSUE TITLE
Use `org.winehq.Wine`

### ISSUE THREAD

**Issue by TheEvilSkeleton** [flathub/com.fightcade.Fightcade#80]
[WINE Flatpak](https://github.com/flathub/org.winehq.Wine) was recently released. Perhaps we can try using this one over `com.fightcade.Fightcade.Wine`?

I recently opened an MR[[1]] in the Lutris Flatpak. We could base it on that.

[1]: https://github.com/flathub/net.lutris.Lutris/pull/212

**Comment by Pobega** [comment]
I don't know if it makes sense for us to track the regular Wine Flatpak; we purposely track a pretty outdated version of Wine (currently Wine 5.0.3) since that's what's recommended and tested by the Fightcade developers. From what I've seen there are usually issues with newer wine versions (for example controller support broke for a lot of people running wine 7) and I'd like to avoid the headaches of not having control of the wine version as that was one of my reasons for wanting to sandbox Fightcade in the first place.

I'm definitely open to it so long as someone can tell me what we'd realistically get out of tracking a newer wine? Wine is only really used for running FBNeo which very rarely changes so compatibility with that isn't an issue.

If anyone is interested in doing the initial work they can throw up a PR and I'd be willing to merge it into the beta branch.

**Comment by TheEvilSkeleton** [comment]
I see, and I completely understand that.

My main intention (which I apologize I haven't explained earlier) is to avoid using the user's storage as much as possible. By using `org.winehq.Wine`, we know we are tracking a "standard" build of WINE Flatpak. By standard, I mean one that is being reused by other applications.

Flatpak deduplicates data, however it depends on the hash of each file. Let's imagine that Fightcade and Lutris use the same `org.winehq.Wine` Flatpak as a base application. If I install Lutris and Fightcade, they will use the exact same WINE. Since Flatpak deduplicates data, WINE won't be using twice the storage, but only once, despite two applications using it. And this gets more and more efficient the more application that require WINE are installed.

So that was the incentive goal, really.

**Comment by Pobega** [comment]
Right, I guess I was unclear -- I understand the advantages from the Flatpak side, I was more so curious if there were advantages from the Wine side. I guess I misinterpreted your intention in opening this issue as I occasionally get requests to update `org.fightcade.Fightcade.Wine` to a more current version, but now I see it's from the Flatpak maintainership side.

Tracking `org.winehq.Wine` is still something I'd definitely consider but as mentioned this specific application (Fightcade) uses Wine to run only two binaries that don't change very often. Until Wine 5.0.3 breaks in some way I'd feel more comfortable tracking the version of Wine recommended by the upstream devs and also avoid the extra headaches that tracking a moving Wine target could cause.

I think it makes sense for the beta branch though, so if a basic PR were to be thrown up I could merge it in there. I could handle the compatibility issues (Fightcade makes a lot of assumptions about the host system and has a lot of weird hacks to get things to play together nicely in the sandbox,) I'd just need whatever the manifest changes to get `org.winehq.Wine` into `com.fightcade.Fightcade` would be.

**Comment by TheEvilSkeleton** [comment]
I wonder if we can add multiple base apps. This Flatpak seems to be using the Electron one:

https://github.com/flathub/com.fightcade.Fightcade/blob/dd3b77a6ec79dfb19cb9dd55acd62b95006d79c0/com.fightcade.Fightcade.yml#L2-L3

I'll ask the Flatpak Matrix.

**Comment by Pobega** [comment]
Yeah, that's what I wasn't sure about otherwise I would've just copied your Lutris PR. Let me know what you hear back.

**Comment by TheEvilSkeleton** [comment]
Someone's response was "I don't think" and this came from a contributor. How long does it take for the Fightcade Flatpak to build usually? If the time is short, I could try it locally.

**Comment by Pobega** [comment]
Without Wine it's about five minutes

**Comment by Pobega** [comment]
@TheEvilSkeleton Three years later but I got around to doing this, would you mind taking a look at the PR? I'll  just merge it (since it's going into beta) but if you have any feedback or see anything I missed that would be great.

https://github.com/flathub/com.fightcade.Fightcade/pull/153
